common/desktop-exports

```diff
@@ -90,7 +90,7 @@ export LIBVA_DRIVERS_PATH=$RUNTIME/usr/lib/$ARCH/dri
 # /var/lib/snapd/lib/gl that needs to be in LD_LIBRARY_PATH
 # Without that OpenGL using apps do not work with the nVidia drivers.
 # Ref.: https://bugs.launchpad.net/snappy/+bug/1588192
-append_dir LD_LIBRARY_PATH /var/lib/snapd/lib/gl
+prepend_dir LD_LIBRARY_PATH /var/lib/snapd/lib/gl
```

Comment:
We *probably* don't need to change this, as there shouldn't be anything in the NVIDIA stack which *overrides* an existing library, so it doesn't matter where in the `LD_LIBRARY_PATH` list it goes.

That said, it's harmless.

Reply:
I suppose this is for future-proofing, in case something shows up here?

Reply:
I don't know if it's more likely to accidentally help in future or accidentally harm in future :woman_shrugging:

Reply:
It's a singular data point, but our performance test snap works only if `/var/lib/snapd/lib/gl` is last in the `LD_LIBRARY_PATH` :grimacing:

Reply:
_actually_:

```bash
find /var/lib/snapd/lib/gl/
/var/lib/snapd/lib/gl/
/var/lib/snapd/lib/gl/vdpau
/var/lib/snapd/lib/gl/vdpau/libvdpau_nvidia.so.440.100
/var/lib/snapd/lib/gl/vdpau/libvdpau_nvidia.so.1
/var/lib/snapd/lib/gl/vdpau/libvdpau_nvidia.so
/var/lib/snapd/lib/gl/libnvoptix.so.440.100
/var/lib/snapd/lib/gl/libnvoptix.so.1
/var/lib/snapd/lib/gl/libnvidia-tls.so.440.100
/var/lib/snapd/lib/gl/libnvidia-rtcore.so.440.100
/var/lib/snapd/lib/gl/libnvidia-ptxjitcompiler.so.440.100
/var/lib/snapd/lib/gl/libnvidia-ptxjitcompiler.so.1
/var/lib/snapd/lib/gl/libnvidia-ptxjitcompiler.so
/var/lib/snapd/lib/gl/libnvidia-opticalflow.so.440.100
/var/lib/snapd/lib/gl/libnvidia-opticalflow.so.1
/var/lib/snapd/lib/gl/libnvidia-opticalflow.so
/var/lib/snapd/lib/gl/libnvidia-opencl.so.440.100
/var/lib/snapd/lib/gl/libnvidia-opencl.so.1
/var/lib/snapd/lib/gl/libnvidia-ml.so.440.100
/var/lib/snapd/lib/gl/libnvidia-ml.so.1
/var/lib/snapd/lib/gl/libnvidia-ml.so
/var/lib/snapd/lib/gl/libnvidia-ifr.so.440.100
/var/lib/snapd/lib/gl/libnvidia-ifr.so.1
/var/lib/snapd/lib/gl/libnvidia-ifr.so
/var/lib/snapd/lib/gl/libnvidia-glvkspirv.so.440.100
/var/lib/snapd/lib/gl/libnvidia-glsi.so.440.100
/var/lib/snapd/lib/gl/libnvidia-glcore.so.440.100
/var/lib/snapd/lib/gl/libnvidia-fbc.so.440.100
/var/lib/snapd/lib/gl/libnvidia-fbc.so.1
/var/lib/snapd/lib/gl/libnvidia-fbc.so
/var/lib/snapd/lib/gl/libnvidia-fatbinaryloader.so.440.100
/var/lib/snapd/lib/gl/libnvidia-encode.so.440.100
/var/lib/snapd/lib/gl/libnvidia-encode.so.1
/var/lib/snapd/lib/gl/libnvidia-encode.so
/var/lib/snapd/lib/gl/libnvidia-eglcore.so.440.100
/var/lib/snapd/lib/gl/libnvidia-compiler.so.440.100
/var/lib/snapd/lib/gl/libnvidia-cfg.so.440.100
/var/lib/snapd/lib/gl/libnvidia-cfg.so.1
/var/lib/snapd/lib/gl/libnvidia-cfg.so
/var/lib/snapd/lib/gl/libnvcuvid.so.440.100
/var/lib/snapd/lib/gl/libnvcuvid.so.1
/var/lib/snapd/lib/gl/libnvcuvid.so
/var/lib/snapd/lib/gl/libcuda.so.440.100
/var/lib/snapd/lib/gl/libcuda.so.1
/var/lib/snapd/lib/gl/libcuda.so
/var/lib/snapd/lib/gl/libGLdispatch.so.0.0.0
/var/lib/snapd/lib/gl/libGLdispatch.so.0
/var/lib/snapd/lib/gl/libGLX.so.0.0.0
/var/lib/snapd/lib/gl/libGLX.so.0
/var/lib/snapd/lib/gl/libGLX_nvidia.so.440.100
/var/lib/snapd/lib/gl/libGLX_nvidia.so.0
/var/lib/snapd/lib/gl/libGLX_indirect.so.0
/var/lib/snapd/lib/gl/libGLESv2_nvidia.so.440.100
/var/lib/snapd/lib/gl/libGLESv2_nvidia.so.2
/var/lib/snapd/lib/gl/libGLESv1_CM_nvidia.so.440.100
/var/lib/snapd/lib/gl/libGLESv1_CM_nvidia.so.1
/var/lib/snapd/lib/gl/libGL.so.1.7.0
/var/lib/snapd/lib/gl/libGL.so.1
/var/lib/snapd/lib/gl/libEGL_nvidia.so.440.100
/var/lib/snapd/lib/gl/libEGL_nvidia.so.0
/var/lib/snapd/lib/gl/libEGL.so.1.1.0
/var/lib/snapd/lib/gl/libEGL.so.1
```

The whole GL stack is exposed here…

Reply:
I've now dropped this `LD_LIBRARY_PATH` modification as the default snapcraft env includes that path via `snapcraft-runner`:

https://github.com/snapcore/snapcraft/blob/bbe83dfd4720b17ac60db462856852231086d1a5/snapcraft/internal/meta/_snap_packaging.py#L462

Reply:
> I don't know if it's more likely to accidentally help in future or accidentally harm in future woman_shrugging

> It's a singular data point, but our performance test snap works only if `/var/lib/snapd/lib/gl` is last in the `LD_LIBRARY_PATH` :grimacing:

So _actually_ our problem was missing `libgles2` on the system, which meant mixed library versions on the path…

As discussed above, it's snapcraft that's putting `SNAP_LIBRARY_PATH` first on `LD_LIBRARY_PATH`, so there's no point doing it in desktop helpers again.